File: test_data/code/python/1624-largest-substring-between-two-equal-characters.py
Python
class Solution:
    def maxLengthBetweenEqualCharacters(self, s: str) -> int:
        char_index = {}
        max_len = -1
        
        for i, char in enumerate(s):
            if char in char_index:
                max_len = max(max_len, i - char_index[char] - 1)
            else:
                char_index[char] = i
        
        return max_len

File: test_data/code/java/1624-largest-substring-between-two-equal-characters.java
class Solution {
    public int maxLengthBetweenEqualCharacters(String s) {
        int maxLen = -1;
        Map<Character, Integer> map = new HashMap<>();
        
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (map.containsKey(c)) {
                maxLen = Math.max(maxLen, i - map.get(c) - 1);
            } else {
                map.put(c, i);
            }
        }
        
        return maxLen;
    }
}